docs/Training_data_sources.md

```diff
@@ -0,0 +1,98 @@
+
+In the notebook `notebooks/Training data analysis.ipynb` I had a look at the extent and effect of the different sources of training data. I wanted to answer 3 broad questions:
+
+1. How much training data comes from each of the 4 tagging sources:
+- EPMC
+- Research Fish
+- Grants via Excel
+- Grants via Prodigy active learning
+2. How well does the 210223 ensemble do for each of these 4 sources?
+3. Does training the models with no EPMC or RF training data contributions improve the model?
+
+This analysis was performed on the following data:
+1. The training data that went into Prodigy : `../data/processed/training_data/210126/training_data.csv`
+2. The final Prodigy merged results (original training + new tags) : `../data/prodigy/merged_tech_grants/merged_tech_grants.jsonl`
+3. Which grants were from the test set for the 210221 model runs: `../data/processed/model_test_results/test_data_210221.csv`
+4. The predictions on all grants from the 210223 Ensemble model (which used the 210221 models): `../data/processed/ensemble/210223/wellcome-grants-awarded-2005-2019_tagged.csv`
+
+# 1. How much training data comes from each of the 4 tagging sources?
+
+The training data is labelled from:
+- Multiple sources: 26
+- RF: 57 (57 'tech')
+- EPMC: 126 (126 'tech')
+- Grants: 485 (138 'tech', 347 'not tech)
+- Prodigy only: 286 (150 'tech', 136 'not tech)
+
+Of the 26 multiple sources only 3 were not in agreement:
+- 3 times the grant description was labelled 'not tech', but RF data labelled as 'tech'
+- 0 times the grant description was labelled 'not tech', but EPMC data labelled as 'tech'
+- 3 times the grant description and RF labels both said tech
+- 13 times the grant description and EPMC labels both said tech
+- 7 times both EPMC and RF labels said tech
+
+# 2. How well does the 210223 ensemble do for each of these 4 sources?
+
+Using the test data (i.e. not any data that went into training the models), the 210223 ensemble model performs differently for data points labelled from the different sources.
+
+- Of the 19 ResearchFish labelled data points (all labelled as tech) **0.632** were correctly labelled as tech when using the grant description.
+- Of the 35 EPMC labelled data points (all labelled as tech) **0.714** were correctly labelled as tech when using the grant description.
```

Comment:
So the performance is worse for EPMC and ResearchFish...

Reply:
yeh. I suppose it speaks to the fact that this is 'hidden tech', perhaps the grant holders didn't think to mention the production of tech in their grant description/it was a byproduct of their grant, so the model had no chance to pick it up from the grant description. I'd have hoped the model has picked up on other language/themes which might make it correlated with a tech kind of grant, but perhaps there wasn't enough to go on to do a great job at this. I think this is an interesting extra metric for evaluation anyway, perhaps wasn't worth including in the training data though.